Sources/NIO/ByteBuffer-core.swift

```diff
@@ -667,6 +667,27 @@ public struct ByteBuffer {
         self._moveWriterIndex(to: 0)
         self._moveReaderIndex(to: 0)
     }
+
+    /// Set both reader index and writer index to `0`. This will reset the state of this `ByteBuffer` to the state
+    /// of a freshly allocated one, if possible without allocations. This is the cheapest way to recycle a `ByteBuffer`
+    /// for a new use-case.
+    ///
+    /// - note: This method will allocate if the underlying storage is referenced by another `ByteBuffer`. Even if an
+    ///         allocation is necessary this will be cheaper as the copy of the storage is elided.
+    ///
+    /// - parameters:
+    ///     - minimumNeededCapacity: The minimum capacity that will be (re)allocated for this buffer
+    public mutating func clear(capacity minimumNeededCapacity: _Capacity) {
```

Comment:
I think this parameter should be named `minimumCapacity` because we are likely to give you a larger buffer, especially if the existing buffer is uniquely referenced.

Reply:
+1

Reply:
@weissi @Lukasa I chose `minimumNeededCapacity` intentionally to align with the existing API from `public func reallocStorage(capacity minimumNeededCapacity: _Capacity)`

Reply:
Changes have been implemented!

Reply:
Also +1 for `minimumCapacity`